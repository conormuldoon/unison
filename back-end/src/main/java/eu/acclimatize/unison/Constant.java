package eu.acclimatize.unison;

import org.springframework.http.HttpStatus;

/**
 * 
 * A class that defines constants.
 *
 */
public final class Constant {

	/**
	 * A private constructor to prevent constant from being instantiated by other
	 * classes.
	 */
	private Constant() {

	}

	/**
	 * The link name and mapping suffix for cloudiness.
	 */
	public static final String CLOUDINESS = "cloudiness";

	/**
	 * The link name and mapping suffix for cloud level.
	 */
	public static final String CLOUD_LEVEL = "cloudLevel";

	/**
	 * The link name and mapping suffix for dew point.
	 */
	public static final String DEW_POINT = "dewPoint";

	/**
	 * The link name and mapping suffix for global radiation.
	 */
	public static final String GLOBAL_RADIATION = "globalRadiation";

	/**
	 * The link name and mapping suffix for fog.
	 */
	public static final String FOG = "fog";

	/**
	 * The link name and mapping suffix for humidity.
	 */
	public static final String HUMIDITY = "humidity";

	/**
	 * The link property name for precipitation.
	 */
	public static final String PRECIPITATION = "precipitation";

	/**
	 * The link name and mapping suffix for pressure.
	 */
	public static final String PRESSURE = "pressure";

	/**
	 * The link name and mapping suffix for temperature.
	 */
	public static final String TEMPERATURE = "temperature";

	/**
	 * The link name and mapping suffix for wind direction.
	 */
	public static final String WIND_DIRECTION = "windDirection";

	/**
	 * The link name and mapping suffix for wind speed.
	 */
	public static final String WIND_SPEED = "windSpeed";

	/**
	 * The link name and mapping suffix for the API explorer.
	 */
	public static final String EXPLORER = "explorer";

	/**
	 * The path variable for the user name.
	 */
	public static final String USER_NAME = "userName";

	/**
	 * The link name and mapping suffix for a user.
	 */
	public static final String USER = "user";

	/**
	 * The mapping suffix and file name (without extension) for the index.
	 */
	public static final String INDEX = "index";

	/**
	 * The mapping suffix for contains.
	 */
	public static final String CONTAINS = "contains";

	/**
	 * The link name and mapping suffix for the location collection.
	 */
	public static final String LOCATION_COLLECTION = "locationCollection";

	/**
	 * The query string parameter field name, GeoJSON property name, and the JPQL
	 * query parameter name for the location name.
	 */
	public static final String LOCATION_NAME = "name";

	/**
	 * The format for the from date and to date query string parameters.
	 */
	public static final String FORMAT = "d-M-yyyy";

	/**
	 * The query string parameter field name and the JPQL query parameter name for
	 * the from date.
	 */
	public static final String FROM_DATE = "fromDate";

	/**
	 * The query string parameter field name and the JPQL query parameter name for
	 * the to date.
	 */
	public static final String TO_DATE = "toDate";

	/**
	 * The location of the application properties file.
	 */
	public static final String PROPERTIES_FILE = "/application.properties";

	/**
	 * The date/time CSV header text.
	 */
	public static final String FROM_HOUR = "From hour";

	/**
	 * The percentage symbol used in CSV headers.
	 */
	public static final String PERCENTAGE_SYMBOL = "(%)";

	/**
	 * The millimetres symbol used in CSV headers.
	 */
	public static final String MILLIMETRES_SYMBOL = "(mm)";

	/**
	 * The Celsius symbol used in CSV headers.
	 */
	public static final String CELSIUS_SYMBOL = "(Â°C)";

	/**
	 * The default used by
	 * {@link eu.acclimatize.unison.csvcontroller.CSVHeaderItem}.
	 */
	public static final String CSV_HEADER_DEFAULT = "";

	/**
	 * The GeoJSON type field name.
	 */
	public static final String TYPE = "type";

	/**
	 * The date format used by a HARMONIE-AROME API.
	 */
	public static final String HARMONIE_DATE_FORMAT = "yyyy-MM-dd'T'KK:mm:ss'Z'";

	/**
	 * The roll name for an authenticated user.
	 */
	public static final String ROLL_USER = "ROLE_USER";

	private static final String HAS_OWNER_AUTH = ".hasOwner(authentication.name)";

	/**
	 * The predicate that the item's owner is the authenticated user.
	 */
	public static final String OWNED_ITEM = "#ownedItem" + HAS_OWNER_AUTH;

	/**
	 * The predicated that the current item's owner is the authenticated user and
	 * updated item's owner is the authenticated user.
	 */
	public static final String REPLACE_ITEM = "#current" + HAS_OWNER_AUTH + " and #updated" + HAS_OWNER_AUTH;

	/**
	 * The created HTTP status code.
	 */
	public static final int CREATED = HttpStatus.CREATED.value();

	/**
	 * The HTTP location header name.
	 */
	public static final String LOCATION_HEADER = "Location";

	/**
	 * The index file name for the root and explorer directories.
	 */
	public static final String INDEX_FILE = MappingConstant.INDEX + ".html";

	public static final String NAME = "name";

}
