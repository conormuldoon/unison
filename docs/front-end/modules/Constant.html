<!DOCTYPE html><html class="default"><head><meta charSet="utf-8"/><meta http-equiv="x-ua-compatible" content="IE=edge"/><title>Constant | @conormuldoon/unison</title><meta name="description" content="Documentation for @conormuldoon/unison"/><meta name="viewport" content="width=device-width, initial-scale=1"/><link rel="stylesheet" href="../assets/style.css"/><link rel="stylesheet" href="../assets/highlight.css"/><script async src="../assets/search.js" id="search-script"></script></head><body><script>document.body.classList.add(localStorage.getItem("tsd-theme") || "os")</script><header><div class="tsd-page-toolbar"><div class="container"><div class="table-wrap"><div class="table-cell" id="tsd-search" data-base=".."><div class="field"><label for="tsd-search-field" class="tsd-widget search no-caption">Search</label><input type="text" id="tsd-search-field"/></div><ul class="results"><li class="state loading">Preparing search index...</li><li class="state failure">The search index is not available</li></ul><a href="../index.html" class="title">@conormuldoon/unison</a></div><div class="table-cell" id="tsd-widgets"><div id="tsd-filter"><a href="#" class="tsd-widget options no-caption" data-toggle="options">Options</a><div class="tsd-filter-group"><div class="tsd-select" id="tsd-filter-visibility"><span class="tsd-select-label">All</span><ul class="tsd-select-list"><li data-value="public">Public</li><li data-value="protected">Public/Protected</li><li data-value="private" class="selected">All</li></ul></div> <input type="checkbox" id="tsd-filter-inherited" checked/><label class="tsd-widget" for="tsd-filter-inherited">Inherited</label><input type="checkbox" id="tsd-filter-externals" checked/><label class="tsd-widget" for="tsd-filter-externals">Externals</label></div></div><a href="#" class="tsd-widget menu no-caption" data-toggle="menu">Menu</a></div></div></div></div><div class="tsd-page-title"><div class="container"><ul class="tsd-breadcrumb"><li><a href="../modules.html">@conormuldoon/unison</a></li><li><a href="Constant.html">Constant</a></li></ul><h1>Module Constant </h1></div></div></header><div class="container container-main"><div class="row"><div class="col-8 col-content"><section class="tsd-panel-group tsd-index-group"><h2>Index</h2><section class="tsd-panel tsd-index-panel"><div class="tsd-index-content"><section class="tsd-index-section "><h3>Variables</h3><ul class="tsd-index-list"><li class="tsd-kind-variable tsd-parent-kind-module"><a href="Constant.html#CL" class="tsd-kind-icon">CL</a></li><li class="tsd-kind-variable tsd-parent-kind-module"><a href="Constant.html#FORMAT" class="tsd-kind-icon">FORMAT</a></li><li class="tsd-kind-variable tsd-parent-kind-module"><a href="Constant.html#HIGH" class="tsd-kind-icon">HIGH</a></li><li class="tsd-kind-variable tsd-parent-kind-module"><a href="Constant.html#HPER" class="tsd-kind-icon">HPER</a></li><li class="tsd-kind-variable tsd-parent-kind-module"><a href="Constant.html#LOW" class="tsd-kind-icon">LOW</a></li><li class="tsd-kind-variable tsd-parent-kind-module"><a href="Constant.html#LPER" class="tsd-kind-icon">LPER</a></li><li class="tsd-kind-variable tsd-parent-kind-module"><a href="Constant.html#MEDIAN" class="tsd-kind-icon">MEDIAN</a></li><li class="tsd-kind-variable tsd-parent-kind-module"><a href="Constant.html#MEDIUM" class="tsd-kind-icon">MEDIUM</a></li><li class="tsd-kind-variable tsd-parent-kind-module"><a href="Constant.html#PRECIP" class="tsd-kind-icon">PRECIP</a></li><li class="tsd-kind-variable tsd-parent-kind-module"><a href="Constant.html#SELF" class="tsd-kind-icon">SELF</a></li></ul></section></div></section></section><section class="tsd-panel-group tsd-member-group "><h2>Variables</h2><section class="tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-module"><a id="CL" class="tsd-anchor"></a><h3 class="tsd-anchor-link"><span class="tsd-flag ts-flagConst">Const</span> CL<a href="#CL" aria-label="Permalink" class="tsd-anchor-icon"><svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-link" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"></path><path d="M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5"></path><path d="M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5"></path></svg></a></h3><div class="tsd-signature tsd-kind-icon">CL<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">&quot;Cloud Level&quot;</span><span class="tsd-signature-symbol"> = &quot;Cloud Level&quot;</span></div><aside class="tsd-sources"><ul><li>Defined in <a href="https://github.com/conormuldoon/unison/blob/d337f3a3/front-end/src/Constant.ts#L12">src/Constant.ts:12</a></li></ul></aside></section><section class="tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-module"><a id="FORMAT" class="tsd-anchor"></a><h3 class="tsd-anchor-link"><span class="tsd-flag ts-flagConst">Const</span> FORMAT<a href="#FORMAT" aria-label="Permalink" class="tsd-anchor-icon"><svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-link" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"></path><path d="M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5"></path><path d="M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5"></path></svg></a></h3><div class="tsd-signature tsd-kind-icon">FORMAT<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">&quot;D-M-YYYY&quot;</span><span class="tsd-signature-symbol"> = &quot;D-M-YYYY&quot;</span></div><aside class="tsd-sources"><ul><li>Defined in <a href="https://github.com/conormuldoon/unison/blob/d337f3a3/front-end/src/Constant.ts#L2">src/Constant.ts:2</a></li></ul></aside></section><section class="tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-module"><a id="HIGH" class="tsd-anchor"></a><h3 class="tsd-anchor-link"><span class="tsd-flag ts-flagConst">Const</span> HIGH<a href="#HIGH" aria-label="Permalink" class="tsd-anchor-icon"><svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-link" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"></path><path d="M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5"></path><path d="M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5"></path></svg></a></h3><div class="tsd-signature tsd-kind-icon">HIGH<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">&quot;High&quot;</span><span class="tsd-signature-symbol"> = &quot;High&quot;</span></div><aside class="tsd-sources"><ul><li>Defined in <a href="https://github.com/conormuldoon/unison/blob/d337f3a3/front-end/src/Constant.ts#L10">src/Constant.ts:10</a></li></ul></aside></section><section class="tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-module"><a id="HPER" class="tsd-anchor"></a><h3 class="tsd-anchor-link"><span class="tsd-flag ts-flagConst">Const</span> HPER<a href="#HPER" aria-label="Permalink" class="tsd-anchor-icon"><svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-link" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"></path><path d="M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5"></path><path d="M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5"></path></svg></a></h3><div class="tsd-signature tsd-kind-icon">HPER<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">&quot;80ᵗʰ percentile&quot;</span><span class="tsd-signature-symbol"> = &quot;80\u1d57\u02b0 percentile&quot;</span></div><aside class="tsd-sources"><ul><li>Defined in <a href="https://github.com/conormuldoon/unison/blob/d337f3a3/front-end/src/Constant.ts#L5">src/Constant.ts:5</a></li></ul></aside></section><section class="tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-module"><a id="LOW" class="tsd-anchor"></a><h3 class="tsd-anchor-link"><span class="tsd-flag ts-flagConst">Const</span> LOW<a href="#LOW" aria-label="Permalink" class="tsd-anchor-icon"><svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-link" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"></path><path d="M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5"></path><path d="M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5"></path></svg></a></h3><div class="tsd-signature tsd-kind-icon">LOW<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">&quot;Low&quot;</span><span class="tsd-signature-symbol"> = &quot;Low&quot;</span></div><aside class="tsd-sources"><ul><li>Defined in <a href="https://github.com/conormuldoon/unison/blob/d337f3a3/front-end/src/Constant.ts#L8">src/Constant.ts:8</a></li></ul></aside></section><section class="tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-module"><a id="LPER" class="tsd-anchor"></a><h3 class="tsd-anchor-link"><span class="tsd-flag ts-flagConst">Const</span> LPER<a href="#LPER" aria-label="Permalink" class="tsd-anchor-icon"><svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-link" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"></path><path d="M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5"></path><path d="M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5"></path></svg></a></h3><div class="tsd-signature tsd-kind-icon">LPER<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">&quot;20ᵗʰ percentile&quot;</span><span class="tsd-signature-symbol"> = &quot;20\u1d57\u02b0 percentile&quot;</span></div><aside class="tsd-sources"><ul><li>Defined in <a href="https://github.com/conormuldoon/unison/blob/d337f3a3/front-end/src/Constant.ts#L4">src/Constant.ts:4</a></li></ul></aside></section><section class="tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-module"><a id="MEDIAN" class="tsd-anchor"></a><h3 class="tsd-anchor-link"><span class="tsd-flag ts-flagConst">Const</span> MEDIAN<a href="#MEDIAN" aria-label="Permalink" class="tsd-anchor-icon"><svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-link" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"></path><path d="M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5"></path><path d="M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5"></path></svg></a></h3><div class="tsd-signature tsd-kind-icon">MEDIAN<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">&quot;Median&quot;</span><span class="tsd-signature-symbol"> = &quot;Median&quot;</span></div><aside class="tsd-sources"><ul><li>Defined in <a href="https://github.com/conormuldoon/unison/blob/d337f3a3/front-end/src/Constant.ts#L6">src/Constant.ts:6</a></li></ul></aside></section><section class="tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-module"><a id="MEDIUM" class="tsd-anchor"></a><h3 class="tsd-anchor-link"><span class="tsd-flag ts-flagConst">Const</span> MEDIUM<a href="#MEDIUM" aria-label="Permalink" class="tsd-anchor-icon"><svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-link" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"></path><path d="M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5"></path><path d="M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5"></path></svg></a></h3><div class="tsd-signature tsd-kind-icon">MEDIUM<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">&quot;Medium&quot;</span><span class="tsd-signature-symbol"> = &quot;Medium&quot;</span></div><aside class="tsd-sources"><ul><li>Defined in <a href="https://github.com/conormuldoon/unison/blob/d337f3a3/front-end/src/Constant.ts#L9">src/Constant.ts:9</a></li></ul></aside></section><section class="tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-module"><a id="PRECIP" class="tsd-anchor"></a><h3 class="tsd-anchor-link"><span class="tsd-flag ts-flagConst">Const</span> PRECIP<a href="#PRECIP" aria-label="Permalink" class="tsd-anchor-icon"><svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-link" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"></path><path d="M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5"></path><path d="M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5"></path></svg></a></h3><div class="tsd-signature tsd-kind-icon">PRECIP<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">&quot;Precipitation&quot;</span><span class="tsd-signature-symbol"> = &quot;Precipitation&quot;</span></div><aside class="tsd-sources"><ul><li>Defined in <a href="https://github.com/conormuldoon/unison/blob/d337f3a3/front-end/src/Constant.ts#L11">src/Constant.ts:11</a></li></ul></aside></section><section class="tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-module"><a id="SELF" class="tsd-anchor"></a><h3 class="tsd-anchor-link"><span class="tsd-flag ts-flagConst">Const</span> SELF<a href="#SELF" aria-label="Permalink" class="tsd-anchor-icon"><svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-link" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"></path><path d="M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5"></path><path d="M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5"></path></svg></a></h3><div class="tsd-signature tsd-kind-icon">SELF<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">&quot;self&quot;</span><span class="tsd-signature-symbol"> = &quot;self&quot;</span></div><aside class="tsd-sources"><ul><li>Defined in <a href="https://github.com/conormuldoon/unison/blob/d337f3a3/front-end/src/Constant.ts#L14">src/Constant.ts:14</a></li></ul></aside></section></section></div><div class="col-4 col-menu menu-sticky-wrap menu-highlight"><nav class="tsd-navigation primary"><ul><li class=""><a href="../modules.html">Modules</a></li><li class=" tsd-kind-module"><a href="ARLocationComponent.html">ARLocation<wbr/>Component</a></li><li class=" tsd-kind-module"><a href="ChartComponent.html">Chart<wbr/>Component</a></li><li class=" tsd-kind-module"><a href="ChartPopup.html">Chart<wbr/>Popup</a></li><li class="current tsd-kind-module"><a href="Constant.html">Constant</a></li><li class=" tsd-kind-module"><a href="DateSelector.html">Date<wbr/>Selector</a></li><li class=" tsd-kind-module"><a href="LeafletMap.html">Leaflet<wbr/>Map</a></li><li class=" tsd-kind-module"><a href="LocationForm.html">Location<wbr/>Form</a></li><li class=" tsd-kind-module"><a href="RemoveComponent.html">Remove<wbr/>Component</a></li><li class=" tsd-kind-module"><a href="TabsComponent.html">Tabs<wbr/>Component</a></li><li class=" tsd-kind-module"><a href="Unison.html">Unison</a></li><li class=" tsd-kind-module"><a href="Util.html">Util</a></li></ul></nav><nav class="tsd-navigation secondary menu-sticky"><ul><li class="tsd-kind-variable tsd-parent-kind-module"><a href="Constant.html#CL" class="tsd-kind-icon">CL</a></li><li class="tsd-kind-variable tsd-parent-kind-module"><a href="Constant.html#FORMAT" class="tsd-kind-icon">FORMAT</a></li><li class="tsd-kind-variable tsd-parent-kind-module"><a href="Constant.html#HIGH" class="tsd-kind-icon">HIGH</a></li><li class="tsd-kind-variable tsd-parent-kind-module"><a href="Constant.html#HPER" class="tsd-kind-icon">HPER</a></li><li class="tsd-kind-variable tsd-parent-kind-module"><a href="Constant.html#LOW" class="tsd-kind-icon">LOW</a></li><li class="tsd-kind-variable tsd-parent-kind-module"><a href="Constant.html#LPER" class="tsd-kind-icon">LPER</a></li><li class="tsd-kind-variable tsd-parent-kind-module"><a href="Constant.html#MEDIAN" class="tsd-kind-icon">MEDIAN</a></li><li class="tsd-kind-variable tsd-parent-kind-module"><a href="Constant.html#MEDIUM" class="tsd-kind-icon">MEDIUM</a></li><li class="tsd-kind-variable tsd-parent-kind-module"><a href="Constant.html#PRECIP" class="tsd-kind-icon">PRECIP</a></li><li class="tsd-kind-variable tsd-parent-kind-module"><a href="Constant.html#SELF" class="tsd-kind-icon">SELF</a></li></ul></nav></div></div></div><footer class="with-border-bottom"><div class="container"><h2>Legend</h2><div class="tsd-legend-group"><ul class="tsd-legend"><li class="tsd-kind-namespace"><span class="tsd-kind-icon">Namespace</span></li><li class="tsd-kind-variable"><span class="tsd-kind-icon">Variable</span></li><li class="tsd-kind-function"><span class="tsd-kind-icon">Function</span></li><li class="tsd-kind-type-alias"><span class="tsd-kind-icon">Type alias</span></li></ul><ul class="tsd-legend"><li class="tsd-kind-interface"><span class="tsd-kind-icon">Interface</span></li></ul></div><h2>Settings</h2><p>Theme <select id="theme"><option value="os">OS</option><option value="light">Light</option><option value="dark">Dark</option></select></p></div></footer><div class="container tsd-generator"><p>Generated using <a href="https://typedoc.org/" target="_blank">TypeDoc</a></p></div><div class="overlay"></div><script src="../assets/main.js"></script></body></html>