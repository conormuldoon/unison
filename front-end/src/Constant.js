
export const FORMAT = "D/M/YYYY";

export const API = "/apiacc";

export const LPER = "20\u1d57\u02b0 percentile";
export const HPER = "80\u1d57\u02b0 percentile";
export const MEDIAN = "Median";

export const LOW = "Low";
export const MEDIUM = "Medium";
export const HIGH = "High";
export const PRECIP = "Precipitation";
export const CL = "Cloud Level";
//export const API = "";

// Map coordinates
//export const CENTRE_LAT = 53.35014;
//export const CENTRE_LON = -6.266155;

// Oslo
export const CENTRE_LAT = 59.922326;
export const CENTRE_LON = 10.751560;
