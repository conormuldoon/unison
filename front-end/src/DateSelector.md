DateSelector example:

```js
import {TODAY} from './Util';

<div style={{"width":"80vw"}}>
    <DateSelector label="Select date" dateValue={TODAY} />
</div>
```