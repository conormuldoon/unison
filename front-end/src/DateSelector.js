import React from 'react';

import DayPickerInput from 'react-day-picker/DayPickerInput';
import 'react-day-picker/lib/style.css';

import  {
  formatDate,
  parseDate,
} from 'react-day-picker/moment';

import {FORMAT} from './Constant';

/**
 * A component for selecting a date.
 * 
 */
export default function DateSelector(props){


    return (
      <div>
        <p>{props.label}</p>
        <DayPickerInput

          format={FORMAT}
          formatDate={formatDate}
          parseDate={parseDate}
          value={props.dateValue}
          onDayChange={props.handleDayChange}


        />
      </div>
    );

}
