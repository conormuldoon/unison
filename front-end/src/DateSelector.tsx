import React from 'react';
import DayPickerInput from 'react-day-picker/DayPickerInput';
import PropTypes from 'prop-types';
import 'react-day-picker/lib/style.css';
import { formatDate, parseDate } from 'react-day-picker/moment';

import { FORMAT } from './Constant';

export interface SelectorProps {

  /**
   * Tha label dispalyed above the date selector.
   */
  label: string;

  /**
   * The initial value of the date selector.
   */
  dateValue: Date;

  /**
   * A callback invoked when the selected date is changed.
   */
  handleDayChange: (day: Date) => void;
}
/**
 * Creates a component for selecting a date.
 * 
 */
function DateSelector({ label, dateValue, handleDayChange }: SelectorProps): JSX.Element {


  return (
    <div>
      <p>{label}</p>
      <DayPickerInput

        format={FORMAT}
        formatDate={formatDate}
        parseDate={parseDate}
        value={dateValue}
        onDayChange={handleDayChange}


      />
    </div>
  );

}

DateSelector.propTypes = {

  /** The label to display over the selector. */
  label: PropTypes.string.isRequired,

  /** The initial date value to display. */
  dateValue: PropTypes.string.isRequired,

  /** A callback function called when the date is changed. */
  handleDayChange: PropTypes.func.isRequired,

}



export default DateSelector;