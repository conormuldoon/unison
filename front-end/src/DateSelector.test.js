import React from 'react';
import ReactDOM from 'react-dom';
import DateSelector from './DateSelector';
import {FORMAT} from './Constant';

import { render} from "react-testing-library";

import  {
  formatDate
} from 'react-day-picker/moment';



const dateSelector=<DateSelector label={'Test Label'} dateValue={'23/10/2019'} handleDayChange={()=>{}}/>;

it('renders without crashing', () => {
  const div = document.createElement('div');


  ReactDOM.render(dateSelector, div);
  ReactDOM.unmountComponentAtNode(div);
});



  it('mathes snapshot', () => {


      const {container} = render(dateSelector);

      expect(container).toMatchSnapshot();

  });
