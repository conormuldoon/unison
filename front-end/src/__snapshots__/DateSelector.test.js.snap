// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mathes snapshot 1`] = `
<div>
  <div>
    <p>
      Test Label
    </p>
    <div
      class="DayPickerInput"
    >
      <input
        placeholder="YYYY-M-D"
        value="11/10/2019"
      />
    </div>
  </div>
</div>
`;
